src/core/mod.rs

```diff
@@ -135,6 +139,128 @@ pub fn check_fault_injection(fault_name: &str) -> bool {
+    Epoch21 = 0x0201,
+}
+
+impl PartialOrd for StacksEpochId {
+    fn partial_cmp(&self, other: &StacksEpochId) -> Option<Ordering> {
+        (*self as u32).partial_cmp(&(*other as u32))
+    }
+}
+
+impl Ord for StacksEpochId {
+    fn cmp(&self, other: &StacksEpochId) -> Ordering {
+        (*self as u32).cmp(&(*other as u32))
+    }
+}
+
+impl TryFrom<u32> for StacksEpochId {
+    type Error = &'static str;
+
+    fn try_from(value: u32) -> Result<StacksEpochId, Self::Error> {
+        match value {
+            x if x == StacksEpochId::Epoch10 as u32 => Ok(StacksEpochId::Epoch10),
+            x if x == StacksEpochId::Epoch20 as u32 => Ok(StacksEpochId::Epoch20),
+            x if x == StacksEpochId::Epoch21 as u32 => Ok(StacksEpochId::Epoch21),
+            _ => Err("Invalid epoch"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct StacksEpoch {
+    pub epoch_id: StacksEpochId,
+    pub start_height: u64,
+    pub end_height: u64,
+}
+
+impl StacksEpoch {
+    #[cfg(test)]
+    pub fn unit_test(first_burnchain_height: u64) -> Vec<StacksEpoch> {
+        vec![
+            StacksEpoch {
+                epoch_id: StacksEpochId::Epoch10,
+                start_height: 0,
+                end_height: first_burnchain_height,
+            },
+            StacksEpoch {
+                epoch_id: StacksEpochId::Epoch20,
+                start_height: first_burnchain_height,
+                end_height: i64::MAX as u64,
+            },
+        ]
+    }
+
+    pub fn all(first_burnchain_height: u64, epoch_2_1_block_height: u64) -> Vec<StacksEpoch> {
+        vec![
+            StacksEpoch {
+                epoch_id: StacksEpochId::Epoch10,
+                start_height: 0,
+                end_height: first_burnchain_height,
+            },
+            StacksEpoch {
+                epoch_id: StacksEpochId::Epoch20,
+                start_height: first_burnchain_height,
+                end_height: epoch_2_1_block_height,
+            },
+            StacksEpoch {
+                epoch_id: StacksEpochId::Epoch21,
+                start_height: epoch_2_1_block_height,
+                end_height: i64::MAX as u64,
+            },
+        ]
+    }
+}
+
+// StacksEpochs are ordered by start block height
+impl PartialOrd for StacksEpoch {
+    fn partial_cmp(&self, other: &StacksEpoch) -> Option<Ordering> {
+        self.start_height.partial_cmp(&other.start_height)
+    }
+}
+
+impl Ord for StacksEpoch {
+    fn cmp(&self, other: &StacksEpoch) -> Ordering {
+        self.start_height.cmp(&other.start_height)
+    }
+}
+
+pub const STACKS_EPOCHS_MAINNET: &[StacksEpoch] = &[
+    StacksEpoch {
+        epoch_id: StacksEpochId::Epoch10,
+        start_height: 0,
+        end_height: BITCOIN_MAINNET_FIRST_BLOCK_HEIGHT,
+    },
+    StacksEpoch {
+        epoch_id: StacksEpochId::Epoch20,
+        start_height: BITCOIN_MAINNET_FIRST_BLOCK_HEIGHT,
+        end_height: STACKS_2_0_LAST_BLOCK_TO_PROCESS,
```

Comment:
If `end_height` is non-inclusive, would it make sense to have `STACKS_2_0_LAST_BLOCK_TO_PROCESS + 1` here?

Reply:
Fixed